### PR TITLE
add 'with_priority()' to handle interlaced request matching conditions

### PR DESCRIPTION
**Pull request by beltram**
Hi @LukeMathWalker,
this is something from the original Wiremock I wanted to bring in wiremock-rs for a long time.  
Let me know what you think

### CODE REVIEW

**Comment by LukeMathWalker** [comment]
This looks good!
I think we need to really make sure the documentation is clear though - what happens when two mocks have the same level of priority but interlacing conditions?

**Comment by beltram** [comment]
Indeed.  
I improved tests to enlight this and documented this on the method : `when 2 mocks have the same priority it fallbacks to insertion order` (same behaviour as original Wiremock btw).  
I was relunctant in repeating this in `lib.rs` because IMO this is a rather advanced usage of wiremock and could be confusing for newcomers

File: src/mock_set.rs [hunk 1]
```diff
@@ -48,6 +48,8 @@ impl MountedMockSet {
     pub(crate) async fn handle_request(&mut self, request: Request) -> (Response, Option<Delay>) {
         debug!("Handling request.");
         let mut response_template: Option<ResponseTemplate> = None;
+        self.mocks
+            .sort_by(|(a, _), (b, _)| a.specification.priority.cmp(&b.specification.priority));
```
Comment (95th) [inline src/mock_set.rs:52]:
```suggestion
            .sort_by_key(|(m, _)| m.specification.priority);
```

Comment (beltram) [reply src/mock_set.rs]: TIL ! Thanks :)

**Comment by LukeMathWalker** [comment]
Merged and released!

**Pull request merged**